test/actions/surveys.js

```diff
@@ -6,117 +6,66 @@ import configureStore from 'redux-mock-store';
 import thunk from 'redux-thunk';
 
 import * as actions from 'src/actions';
-import { SURVEYS_ACTION, NEW_SURVEY_ID, NEW_SURVEY } from 'src/constants';
+import { SURVEYS_ACTION, NEW_SURVEY_ID, NEW_SURVEY_BODY } from 'src/constants';
 import { mockSurveys, mockSurveysMetadata } from 'test/utils';
 
 const mockStore = configureStore([thunk]);
 
 describe('surveys actions', () => {
-  it('should dispatch an action to fetch metadata when it is empty', () => {
+  it('should dispatch an action to fetch metadata', () => {
     const store = mockStore({});
 
     store.dispatch(actions.fetchSurveysMetadata());
 
-    // TODO(ayushbhagat): When the action makes a request to the backend, change the expectedActions
+    // TODO(ayushbhagat): When the action makes a request to the backend, change the expectedAction
     // to mock that request.
-    const expectedActions = {
+    const expectedAction = {
       type: SURVEYS_ACTION.CHANGE_SURVEYS_METADATA,
       payload: {
-        surveysMetadata: Immutable.fromJS(mockSurveysMetadata),
+        surveysMetadata: mockSurveysMetadata,
       },
     };
     const actualActions = store.getActions();
-    expect(actualActions).to.deep.equal([expectedActions]);
-  });
-
-  it('should not dispatch an action to fetch metadata when it is not empty', () => {
-    const initalState = {
-      surveys: Immutable.fromJS({
-        metadata: mockSurveysMetadata,
-      }),
-    };
-    const store = mockStore(initalState);
-
-    store.dispatch(actions.fetchSurveysMetadata());
-
-    const actualActions = store.getActions();
-    expect(actualActions).to.be.empty;
+    expect(actualActions).to.deep.equal([expectedAction]);
   });
 
-  it('should dispatch actions to update cache and change survey when it is not in cache', () => {
+  it('should dispatch action to change an existing survey', () => {
     const store = mockStore({});
 
-    const id = 1;
-    store.dispatch(actions.fetchSurvey(id));
-
-    const expectedAddSurveyToCacheAction = {
-      type: SURVEYS_ACTION.ADD_SURVEY_TO_CACHE,
-      payload: {
-        id: id.toString(),
-        survey: Immutable.fromJS(mockSurveys[id]),
-      },
-    };
-    const expectedChangeSurveyAction = {
-      type: SURVEYS_ACTION.CHANGE_SURVEY,
-      payload: {
-        id: id.toString(),
-      },
-    };
-    const actualActions = store.getActions();
-    expect(actualActions).to.deep.equal(
-        [expectedAddSurveyToCacheAction, expectedChangeSurveyAction]);
-  });
-
-  it('should only dispatch action to change survey if survey is in cache', () => {
-    const initalState = {
-      surveys: Immutable.fromJS({
-        cache: mockSurveys,
-      }),
-    };
-    const store = mockStore(initalState);
-
-    const id = 1;
+    const id = '1';
```

Comment:
So the reason why I made it `'new'` was because I wanted to ensure that the route for the page where you can edit a survey with an id `id` was `/survey/id`. And since the route for the page where you can create a survey is called `/survey/new`, I made the id `'new'`. But now that I think about it, I don't think having the same `id` as one in the route provides any benefit for a survey that hasn't been persisted. So I'll change it.